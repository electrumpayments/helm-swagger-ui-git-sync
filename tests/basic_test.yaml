suite: Basic templating
values:
  - ../values.yaml
release:
  name: swagger-my-app
  namespace: my-app-ns
set:
  git:
    repo_url: git@github.com:swagger-api/swagger-petstore.git
    swagger_path: src/main/resources/openapi.yaml
tests:
  - it: defaults
    asserts:
      - matchSnapshot: {}
  - it: with ssh
    set:
      git.ssh:
        existing_secret_name: my-app-secret
        existing_secret_key: deploy-key
    template: deployment.yaml
    documentIndex: 0
    asserts:
      - matchSnapshot:  # secret must be mounted and configured for git-sync
          path: spec.template.spec.containers[1]
      - matchSnapshot:  # secret volume must exist
          path: spec.template.spec.volumes
  - it: custom git config
    set:
      git:
        ref: dev
        refresh_period: 12h
        swagger_path: my/cool/path.yaml
    template: deployment.yaml
    asserts:
      - documentIndex: 0
        matchSnapshot:  # git-sync args defined correctly
          path: spec.template.spec.containers[1].args
      - documentIndex: 0
        matchSnapshot:  # swagger-ui points to the correct path
          path: spec.template.spec.containers[0].env
      - documentIndex: 1
        matchSnapshot:
          path: data
