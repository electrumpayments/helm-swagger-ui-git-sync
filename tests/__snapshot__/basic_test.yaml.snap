additional envs:
  1: |
    - name: SWAGGER_JSON
      value: /git/current/src/main/resources/openapi.yaml
    - name: DEFAULT_MODEL_EXPAND_DEPTH
      value: "5"
    - name: TRY_IT_OUT_ENABLED
      valueFrom:
        configMapKeyRef:
          key: try_it_out
          name: my-app-cm
custom git config:
  1: |
    - --repo=git@github.com:swagger-api/swagger-petstore.git
    - --ref=dev
    - --depth=1
    - --period=12h
    - --link=current
    - --root=/git
    - --verbose=6
  2: |
    - name: SWAGGER_JSON
      value: /git/current/my/cool/path.yaml
defaults:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        io.electrum/app: swagger-my-app
      name: swagger-my-app
    spec:
      replicas: 1
      selector:
        matchLabels:
          io.electrum/app: swagger-my-app
      template:
        metadata:
          labels:
            io.electrum/app: swagger-my-app
        spec:
          containers:
            - env:
                - name: SWAGGER_JSON
                  value: /git/current/src/main/resources/openapi.yaml
              image: swaggerapi/swagger-ui:v5.17.14
              livenessProbe:
                httpGet:
                  path: /
                  port: http
                periodSeconds: 10
              name: swagger-ui
              ports:
                - containerPort: 8080
                  name: http
              resources:
                limits:
                  memory: 64Mi
                requests:
                  cpu: 20m
                  memory: 64Mi
              startupProbe:
                failureThreshold: 24
                httpGet:
                  path: /
                  port: http
                periodSeconds: 5
              volumeMounts:
                - mountPath: /git
                  name: git-content
                - mountPath: /docker-entrypoint.d/35-git-sync-init.sh
                  name: docker-init
                  subPath: git-sync-init.sh
            - args:
                - --repo=git@github.com:swagger-api/swagger-petstore.git
                - --ref=main
                - --depth=1
                - --period=300s
                - --link=current
                - --root=/git
                - --verbose=6
              image: registry.k8s.io/git-sync/git-sync:v4.0.0
              name: git-sync
              resources:
                limits:
                  memory: 32Mi
                requests:
                  cpu: 20m
                  memory: 32Mi
              startupProbe:
                exec:
                  command:
                    - sh
                    - -c
                    - test -d /git/current
                failureThreshold: 24
                periodSeconds: 5
              volumeMounts:
                - mountPath: /git
                  name: git-content
          securityContext:
            fsGroup: 65533
          volumes:
            - emptyDir: {}
              name: git-content
            - configMap:
                defaultMode: 493
                name: swagger-my-app-docker-init
              name: docker-init
  2: |
    apiVersion: v1
    data:
      git-sync-init.sh: |
        #!/bin/sh
        max_tries=12
        tries=0

        echo "Waiting for swagger file to appear at path: $SWAGGER_JSON"
        until test -f "$SWAGGER_JSON"; do
          echo "Swagger file not present. Waiting for git-sync..."
          tries=$((tries+1))
          if [ $tries -ge $max_tries ]; then
            echo "Giving up after $max_tries tries"
            exit 1
          fi
          sleep 5
        done
        echo "Swagger file present!"
    kind: ConfigMap
    metadata:
      name: swagger-my-app-docker-init
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        io.electrum/app: swagger-my-app
      name: swagger-my-app
    spec:
      ports:
        - name: http
          port: 80
          targetPort: http
      selector:
        io.electrum/app: swagger-my-app
with ssh:
  1: |
    args:
      - --repo=git@github.com:swagger-api/swagger-petstore.git
      - --ref=main
      - --depth=1
      - --period=300s
      - --link=current
      - --root=/git
      - --ssh=true
      - --ssh-key-file=/etc/git-secret/id_rsa
      - --ssh-known-hosts=false
      - --verbose=6
    image: registry.k8s.io/git-sync/git-sync:v4.0.0
    name: git-sync
    resources:
      limits:
        memory: 32Mi
      requests:
        cpu: 20m
        memory: 32Mi
    startupProbe:
      exec:
        command:
          - sh
          - -c
          - test -d /git/current
      failureThreshold: 24
      periodSeconds: 5
    volumeMounts:
      - mountPath: /git
        name: git-content
      - mountPath: /etc/git-secret
        name: ssh-key
        readOnly: true
  2: |
    - emptyDir: {}
      name: git-content
    - name: ssh-key
      secret:
        defaultMode: 256
        items:
          - key: deploy-key
            path: id_rsa
        secretName: my-app-secret
    - configMap:
        defaultMode: 493
        name: swagger-my-app-docker-init
      name: docker-init
